### PR TITLE
Tool gnat project facility

### PR DESCRIPTION
**Pull request by frank-at-adacore**

### CODE REVIEW

**Review by leogermond — CHANGES_REQUESTED** [review]

File: courses/gnat_project_facility/020_building_with_gprbuild.rst [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+------------------------------
+Common Command Line Switches
+------------------------------
+
+.. container:: latex_environment tiny
+
+  .. list-table::
+
+    * - -P *<project file>*
```
[the remark this answers is not in the document]
Comment (frank-at-adacore) [reply courses/gnat_project_facility/020_building_with_gprbuild.rst:130]: Done

File: courses/gnat_project_facility/030_project_properties.rst [hunk 2]
```diff
@@ -0,0 +1,525 @@
...
+
+     :ada:`for Source_Dirs use ("src/**");`
+
+   + :filename:`src` directory and every subdirectory underneath
+
+--------------
+Source Files
+--------------
+
++ Must be at least one **immediate** source file
```
[the remark this answers is not in the document]
Comment (frank-at-adacore) [reply courses/gnat_project_facility/030_project_properties.rst:144]: First bullet was (kind of) - made it more explicit

File: courses/gnat_project_facility/030_project_properties.rst [hunk 3]
```diff
@@ -0,0 +1,525 @@
...
+        project Release is
+           for Object_Dir use "release";
+           ...
+        end Release;
+
++ Only one per project
+
+  + When extending a parent project
+
+    + Child's object directory contains output for source not already compiled in parent
```
[the remark this answers is not in the document]
Comment (frank-at-adacore) [reply courses/gnat_project_facility/030_project_properties.rst:179]: rephrased

File: courses/gnat_project_facility/040_structuring_your_application.rst [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+
++ Use the tick (') for attributes
+
+   .. code:: Ada
+
+      with "foo.gpr";
+      project P is
+         package Compiler is
+            for Default_Switches ("Ada") use 
+               Foo.Compiler'Default_Switches & "-gnatwa";
```
[the remark this answers is not in the document]
Comment (frank-at-adacore) [reply courses/gnat_project_facility/040_structuring_your_application.rst:188]: added a note (it's in the GPR UG)

File: courses/gnat_project_facility/040_structuring_your_application.rst [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+      ...
+   end Clients;
+
+----------------------
+Project Dependencies
+----------------------
+
++ Not unusual for projects to be interdependent
+
+  + Source in project A refers to source in project B, and source in project B refers to source in project A
```
[the remark this answers is not in the document]
Comment (frank-at-adacore) [reply courses/gnat_project_facility/040_structuring_your_application.rst]: Done

File: courses/gnat_project_facility/labs/source/020_building_with_gprbuild/queens.adb [hunk 6]
```diff
@@ -0,0 +1,16 @@
+with Ada.Command_Line;
```
[the remark this answers is not in the document]
Comment (frank-at-adacore) [reply courses/gnat_project_facility/labs/source/020_building_with_gprbuild/queens.adb:1]: All set

**Review by leogermond — DISMISSED** [review]

**Pull request merged**